### ISSUE TITLE
[OptiFine HD U G9 pre30] Glint in enchanted items and blocks don't work properly in some shaders

### ISSUE THREAD

**Issue by javiergcim** [sp614x/optifine#6065]
## Description of Issue
In some shaders (MakeUp, Super Duper Vanilla, maybe others), enchanted objects (in hand or in the world) appear black. This behavior is not present in version 1.16.x

In other shaders (Enhanced Default), the glint has a strange pattern or noise.

With the MakeUp shader the object is affected by the Bloom effect, as if black were actually a very shiny thing, producing a black bloom (signed / unsigned value problem?)

When the glint color is manually set in a shader such as "gbuffers_armor_glint", the set color works correctly.

## Steps to Reproduce
Only one enchanted item needs to be displayed on the screen

## OptiFine Version
OptiFine HD U G9 pre30

## Installation Method
Standalone installer

## Other Installed Mods
None

**Comment by Chris6ix** [comment]
Can confirm this. Using OptiFine HD U G9 pre31 (Standalone installer)
-Enchanted items with Chocapic13 V6 are black (strangely this is only in the overworld...in the nether and end they work fine)
-Enchanted items with BSL v8.0.01 have  a weird "flickering" glint pattern

**Comment by javiergcim** [comment]
I have found that the behavior changes if the type of buffer in which a glint buffer writes is different.

In MakeUp the buffer is of type R11F_G11F_B10F, if it changes to some other type, such as RGBA16, the black is not appreciated, but the glint is not appreciated either.

Hopefully this information is useful.

To change the type used in the buffer, you can edit MakeUp's final.fsh file.

**Comment by VillagerMaster68** [comment]
The glint animation isn’t present in the end or nether and is just black in the overworld

**Comment by Chris6ix** [comment]
> 
> 
> Is this still an issue in the latest preview versions?

Yes. I just tested this with OptiFine HD U G9 pre35 (Standalone installer)
Enchanted Items still have a weird "flickering" glint pattern with BSL v8.0.01
And Enchanted Items with Chocapic V6 are still black (Only in the Overworld)
